front/src/Stores/BiggestAvailableAreaStore.ts

```diff
@@ -1,112 +1,136 @@
-                xEnd: game.offsetWidth,
-                yEnd: game.offsetHeight,
-            };
-        }
+    // map iframes to Box'es and remove ones outside of the game canvas space
+    const iframeBoxes: Box[] = Array.from(document.getElementsByTagName("iframe"))
+        .filter((iframe) => iframe.offsetLeft < game.offsetWidth)
+        .map((iframe) => ({
+            xStart: iframe.offsetLeft,
+            yStart: iframe.offsetTop,
+            xEnd: iframe.offsetLeft + iframe.offsetWidth,
+            yEnd: iframe.offsetTop + iframe.offsetHeight,
+        }));
+
+    console.log(Array.from(document.getElementsByClassName("video-container")));
+    // const videoContainers: Box[] =
+
+    // create vertices arrays and insert game canvas edges
+    const xVertices = [game.offsetLeft, game.offsetWidth];
+    const yVertices = [game.offsetTop, game.offsetHeight];
+
+    // populate with all vertices
+    for (const box of iframeBoxes) {
+        xVertices.push(box.xStart, box.xEnd);
+        yVertices.push(box.yStart, box.yEnd);
+    }
+    xVertices.sort();
+    yVertices.sort();
+
+    // NOTE: cannot use fill() here it makes references to a single array
+    const occupiedSpace: boolean[][] = new Array(xVertices.length);
+    for (let x = 0; x < xVertices.length; x += 1) {
+        occupiedSpace[x] = new Array(yVertices.length);
+    }
 
-        const lastDiv = htmlChildren[htmlChildren.length - 1];
-        // Compute area between top right of the last div and bottom right of window
-        const area1 =
-            (game.offsetWidth - (lastDiv.offsetLeft + lastDiv.offsetWidth)) * (game.offsetHeight - lastDiv.offsetTop);
-
-        // Compute area between bottom of last div and bottom of the screen on whole width
-        const area2 = game.offsetWidth * (game.offsetHeight - (lastDiv.offsetTop + lastDiv.offsetHeight));
-
-        if (area1 < 0 && area2 < 0) {
-            // If screen is full, let's not attempt something foolish and simply center character in the middle.
-            return {
-                xStart: 0,
-                yStart: 0,
-                xEnd: game.offsetWidth,
-                yEnd: game.offsetHeight,
-            };
+    // check if given vertex is a part of something occupying the view
+    for (let x = 0; x < xVertices.length; x += 1) {
+        for (let y = 0; y < yVertices.length; y += 1) {
+            occupiedSpace[x][y] = isPartOfBoxes(xVertices[x], yVertices[y], iframeBoxes);
         }
-        if (area1 <= area2) {
-            return {
-                xStart: 0,
-                yStart: lastDiv.offsetTop + lastDiv.offsetHeight,
-                xEnd: game.offsetWidth,
-                yEnd: game.offsetHeight,
-            };
-        } else {
-            return {
-                xStart: lastDiv.offsetLeft + lastDiv.offsetWidth,
-                yStart: lastDiv.offsetTop,
-                xEnd: game.offsetWidth,
-                yEnd: game.offsetHeight,
-            };
+    }
```

Comment:
What if you were instead iterating on boxes to fill the occupiedSpace area?

Something like:

```
for (const box of iframeBoxes) {
    // The findBoxCoordinates returns the coordinates of the box in the occupiedSpace matrix
    [iStart, iEnd, jStart, jEnd] = findBoxCoordinates(box, xVertices, yVertices);
    for (i=iStart; i<iEnd; i++) {
        for (j=jStart; j<jEnd; j++) {
            occupiedSpace[i][j] = true;
        }

    }
}